JASP-Engine/JASP/R/exploratoryfactoranalysis.R

```diff
@@ -15,850 +15,572 @@
+  if (inherits(pa, "try-error"))                  return(1)
+  if (options$factorMethod == "parallelAnalysis") return(max(1, pa$ncomp))
+  if (options$factorMethod == "eigenValues") {
+    ncomp <- sum(pa$fa.values > options$eigenValuesBox)
+    # I can use stop() because it's caught by the try and the message is put on
+    # on the modelcontainer.
+    if (ncomp == 0)
+      stop(
+        "No components with an eigenvalue > ", options$eigenValuesBox, ". ",
+        "Maximum observed eigenvalue: ", round(max(pa$fa.values), 3)
+      )
+    return(ncomp)
   }
-  
-  ## this is the main object; we add stuff to it and return it
-  results <- list(
-    "citation" = list("Revelle, W. (2015) psych: Procedures for Personality and Psychological Research, Northwestern University,
-		Evanston, Illinois, USA, http://CRAN.R-project.org/package=psych Version = 1.5.8.")
-  )
-  
-  list("results" = results, "dataset" = dataset)
 }
 
 
-# Path diagram:
-.pathDiagramPCAEFA <- function(analysisResults, options, perform, type, oldPlot=NULL){
-  
-  if (!is.null(oldPlot) && !identical(oldPlot[["data"]], "")) {
-    return(oldPlot)
-  }
-  
-  pathDiagram <- list()
-  pathDiagram$title <- "Path Diagram"
-  
-  if (options$rotationMethod == "orthogonal") {
-    Rotation <- options$orthogonalSelector
-  } else {
-    Rotation <- options$obliqueSelector
-  }
-  
-  xName <- ifelse(Rotation == "none", "PC", "RC")
-  
-  pathDiagram$width <- 480
-  if (length(options$variables) < 2) {
-    pathDiagram$height <- 300
-  } else {
-    pathDiagram$height <- 1 + 299 * (length(options$variables)/5)
-  }
-  pathDiagram$custom <- list(width="plotWidthPathDiagram", height="plotHeightPathDiagram")
-  
-  if (is.null(analysisResults) || isTryError(analysisResults)) { 
-    
-    pathDiagram$data <- NULL
-    
+# Output functions ----
+.efaKMOtest <- function(modelContainer, dataset, options, ready) {
+  if (!options[["kmotest"]] || !is.null(modelContainer[["kmotab"]])) return()
+  
+  kmotab <- createJaspTable("Kaiser-Meyer-Olkin test")
+  kmotab$dependOn("kmotest")
+  kmotab$addColumnInfo(name = "col", title = "", type = "string")
+  kmotab$addColumnInfo(name = "val", title = "MSA", type = "number", format = "dp:3")
+  kmotab$position <- -1
+  modelContainer[["kmotab"]] <- kmotab
+  
+  if (!ready) return()
+  kmo <- psych::KMO(dataset)
+  
+  kmotab[["col"]] <- c("Overall MSA\n", .unv(names(kmo$MSAi)))
+  kmotab[["val"]] <- c(kmo$MSA, kmo$MSAi)
+}
+
+.efaBartlett <- function(modelContainer, dataset, options, ready) {
+  if (!options[["bartest"]] || !is.null(modelContainer[["bartab"]])) return()
+  
+  bartab <- createJaspTable("Bartlett's test")
+  bartab$dependOn("bartest")
+  bartab$addColumnInfo(name = "chisq", title = "\u03a7\u00b2", type = "number", format = "dp:3")
+  bartab$addColumnInfo(name = "df", title = "df", type = "number", format = "dp:3")
+  bartab$addColumnInfo(name = "pval", title = "p-value", type = "number", format = "dp:3;p:.001")
```

Comment:
```suggestion
  bartab$addColumnInfo(name = "pval", title = "p", type = "number", format = "dp:3;p:.001")
```
to be in-line with your other table (and most of jasp I think)